src/support/safe_integer.cpp

```diff
@@ -50,13 +50,15 @@ int32_t wasm::toSInteger32(double x) {
 }
 
 bool wasm::isUInteger64(double x) {
-  return !std::signbit(x) && isInteger(x) &&
-         x <= std::numeric_limits<uint64_t>::max();
+  static_assert(std::numeric_limits<double>::is_iec559,
+                "this function only works with IEEE 754 floating point");
+  return !std::signbit(x) && isInteger(x) && x < 0x1p64;
```

Comment:
The main problem with `0x1p64` it's become `18446744073709552000.0` for `double` representation but we need `18446744073709551616` for precise comparision